zetaclient/chains/solana/signer/signer.go

```diff
@@ -163,13 +163,23 @@ func (signer *Signer) TryProcessOutbound(
 		}
 
 	case coin.CoinType_ERC20:
-		withdrawSPLTx, err := signer.prepareWithdrawSPLTx(ctx, cctx, height, logger)
-		if err != nil {
-			logger.Error().Err(err).Msgf("TryProcessOutbound: Fail to sign withdraw spl outbound")
-			return
-		}
+		if cctx.InboundParams.IsCrossChainCall && IsPendingOutboundFromZetaChain(cctx, zetacoreClient) {
```

Comment:
This looks good. The three other cases will go to the `else` path

1.  Normal SPL withdraw
2. Pending revert of a normal SPL deposit
3. Pending revert of a normal SPL deposit and call.